test/e2e/fixtures/then.go

```diff
@@ -134,7 +134,7 @@ func (t *Then) ExpectWorkflowList(listOptions metav1.ListOptions, block func(t *
 	return t
 }
 
-func (t *Then) ExpectWorkflowListRetryOnError(listOptions metav1.ListOptions, block func(t *testing.T, client v1alpha1.WorkflowInterface, listOptions metav1.ListOptions)) *Then {
+func (t *Then) WaitForWorkflowList(listOptions metav1.ListOptions, block func(t *testing.T, client v1alpha1.WorkflowInterface, listOptions metav1.ListOptions)) *Then {
```

Comment:
I think this should be a When, not a Then

Reply:
take a look at 	WaitForWorkflow() for example of implementation

Reply:
> take a look at WaitForWorkflow() for example of implementation

Not clear to me what you suggest here? Do you refer to that I should implement a function similar to `WaitForWorkflow` where we instead wait for a list condition to be true? Or only that you think that `WaitForWorkflowList` should be reimplemented in `When` instead?

Reply:
What I'm suggesting is that we have:

```golang
func (w *When) WaitForWorkflowList(listOptions metav1.ListOptions, condition func(list []*wfv1.Workflow) bool)
```